js/script.js

```diff
@@ -0,0 +1,185 @@
+let allTasks = [];
+let valueInput = '';
+let input = null;
+let content;
+let activeEditTask = null;
+
+const buttonTasksAdd = async () => {
+  if (!valueInput.trim()) { return; }
+
+  allTasks.push({
+    text: valueInput.trim(),
+    isCheck: false,
+  });
+
+  const resp = await fetch('http://localhost:8000/createTask', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'Access-Control-Allow-Origin': '*'
+    },
+    body: JSON.stringify({
+      text: valueInput.trim(),
+      isCheck: false,
+    }),
+  });
+  let result = await resp.json();
+  allTasks  = result.data;
+  valueInput = '';
+  input.value = '';
+  render();
+}
+
+const newLocal = 'change';
+
+window.onload = async function init() {
+  input = document.getElementById('add-task');
+  input.addEventListener('change', updateValue);
+  content = document.getElementById('content-page');
+  const button = document.querySelector('.add__btn');
+  button.addEventListener('click', buttonTasksAdd);
+
+  const resp = await fetch('http://localhost:8000/allTasks', {
+    method: "GET"
+  });
+  let result = await resp.json();
+  allTasks  = result.data;
+  render();
+}
+
+const updateValue = (event) => {
+  valueInput = event.target.value;
+}
+
+const render = () => {
+
+  while (content && content.firstChild) {
+    content.removeChild(content.firstChild);
+  };
+
+  allTasks.sort((a, b) => a.isCheck > b.isCheck ? 1 : a.isCheck < b.isCheck ? -1: 0);
+  allTasks.forEach((item, index) => {
+    const container = document.createElement('div');
+    container.id = `task-${index}`;
+    container.className = 'task__container';
+    
+    const checkbox = document.createElement('input');
+    checkbox.type = 'checkbox';
+    checkbox.checked = item.isCheck;
+    checkbox.onchange = () => {
+      onChangeCheckbox(index);
+    };
+
+    container.appendChild(checkbox);
+
+    if (index === activeEditTask) {
+      const inputTask = document.createElement('input');
+      inputTask.type = 'text';
+      inputTask.value = item.text;
+      inputTask.addEventListener('change', updateTaskText);

```

Comment:
why is the request sent every time the text changes?
That is why you use change and not click